### ISSUE TITLE
MariaDB as packaged by Ubuntu does not prompt for a password on install

### ISSUE THREAD

**Issue by devosb** [nextcloud/documentation#708]
I had difficulty installing Nextcloud due to how MariaDB was configured. See https://github.com/nextcloud/server/issues/8442 for the details. I don't know if the documentation should be changed to

* say don't use MariaDB from the Ubuntu repos, use packages from the MariaDB project -or-
* manually create a root2 database user, and use that user for the Nextcloud database initialization step.

Depending on which option is chosen, the backup and restore instructions may have to change (due to character set and collation settings)

**Comment by c0fe** [comment]
what is your ubuntu and mariadb version?

**Comment by orangepizza** [comment]
IIRC you enter mariadb as root with sudo and can config it.
like `sudo mysql` then you don't need to enter or need root password for it.

**Comment by c0fe** [comment]
@orangepizza not correct. Anyways, I attempted to reproduce the problem that @devosb is having but I couldn't reproduce it. Without knowing other details I cannot assist.

**Comment by devosb** [comment]
@c0fe I am using Ubuntu 16.04.3 LTS x86_64 with MariaDB as packaged by Ubuntu, version 10.0.33-0ubuntu0.16.04.1 (from dpkg -l). Happy to provide details, sorry for the delay. Am I missing something? I would have thought these two details could be obtained from issue https://github.com/nextcloud/server/issues/8442.

@orangepizza Yes, I used `sudo mysql -u root` to add a root2 user. So AFAIKT, your suggestions works to set a password for the root user. However, if I give the password for root to the graphical Installation Wizard, it does not work, which is why I created a root2 user and gave that username and password to the graphical Installation Wizard, and it works.

I would guess that the reason is explained by the message `The password will be ignored.` from `systemctl`. Full details at https://github.com/nextcloud/server/issues/8442.

**Comment by orangepizza** [comment]
@devosb  I never set a password to root and just used `sudo mysql`  to manage, without -u root

**Comment by c0fe** [comment]
@devosb i see the issue now, ok i will make some edits.

I do recommend that you install MariaDB from their repo and not from Ubuntu's. Stick to their stable versions.

**Comment by devosb** [comment]
I do give the Installation Wizard privileges on everything since the documentation says `You will need the root database login, or any administrator login that has permissions to create and modify databases, ...` at https://docs.nextcloud.com/server/13/admin_manual/installation/installation_wizard.html#database-choice-label

The next paragraph goes on to say that `...needs only the special Nextcloud database user, and drops the root dB login.` So once the server is installed, no I do not need to give nextcloud access to the entire database.

So does the installer (either graphically or command line) work if you create a database, as you have suggested, and a nextcloud user, and give that information to the installer?

**Comment by orangepizza** [comment]
@devosb yes, I though most tutorials for nextcloud install do that, for exemple, https://www.howtoforge.com/tutorial/install-nextcloud-server-and-client-on-debian-9/

**Comment by c0fe** [comment]
@orangepizza howtoforge is a bad site for tutorials in my opinion, i found a lot of wrong steps. best is digitalocean.

**Comment by orangepizza** [comment]
@c0fe I didn't read it (I used a non-english tuturial) just used it to show it's possible to use non-root account for nextcloud works and most tutoral don't give root password to nextcloud (who does that?)

**Comment by c0fe** [comment]
I will be making changes to this. @MorrisJobke can you please assign me to this? I will get this done.

**Comment by c0fe** [comment]
So I made changes to this, mainly just adding a note for Ubuntu install that the root password won't have a prompt and you should run the mysql_secure_installation command after installation.

I will check on CentOS shortly as well. I do not know how the Nextcloud devs feel about adding information about using MariaDB repo instead of the distro's own repo.

**Pros:**
* Software on the developer's repo is more up to date compared to the distro's repository in most cases
* May be a quicker method to receive important security updates as well.
* Easier to manage as you do not have to check how other distros handle this.
* Other distros may not even have MariaDB as an option in their own repos.

**Cons:**
* Have to check for changes with each distro.

That being said I think adding a note in there is the best way for now unless I hear otherwise from the devs.

**Comment by joshtrichards** [comment]
This Issue can be closed out - it's not longer relevant and the docs have long ago been updated in any case:

https://docs.nextcloud.com/server/latest/admin_manual/installation/example_ubuntu.html#

**Comment by skjnldsv** [comment]
Docs have been updated. See https://docs.nextcloud.com/server/latest/admin_manual/installation/example_ubuntu.html
